@xen-orchestra/web/src/pages/vm/new.vue

```diff
@@ -643,66 +590,93 @@ const isDiskTemplate = computed(() => {
 //   },
 // })
 
-const defaultSr = computed(() => {
-  const _defaultSr = vmState.pool?.default_SR
-  return vmState.pool && _defaultSr ? getSr(_defaultSr)?.id : ''
-})
-
 const filteredSrs = computed(() => {
   return srs.value.filter(sr => sr.content_type !== 'iso' && sr.physical_usage > 0 && sr.$pool === vmState.pool?.id)
 })
 
-const getVdis = (template: XoVmTemplate) =>
+const getVmTemplateVdis = (template: XoVmTemplate) =>
   (template.template_info?.disks ?? []).map((disk, index) => ({
     name_label: `${vmState?.name || 'disk'}_${index}_${generateRandomString(4)}`,
     name_description: 'Created by XO',
     size: bytesToGiB(disk.size),
-    sr: defaultSr.value || '',
+    sr: defaultSr.value,
   }))
 
-const getExistingDisks = (template: XoVmTemplate) =>
-  template.$VBDs.flatMap(vbd => {
-    const vdi = getVdi(getVbd(vbd)!.VDI)
-    return vdi
-      ? [
-          {
-            id: vdi.id,
-            name_label: vdi.name_label,
-            name_description: vdi.name_description,
-            size: bytesToGiB(vdi.size),
-            sr: getSr(vdi.$SR)?.id || '',
-          },
-        ]
-      : []
-  })
+const getExistingVdis = (template: XoVmTemplate) => {
+  return template.$VBDs.reduce<Vdi[]>((acc, vbdId) => {
+    const _vbd = getVbd(vbdId)
+
+    if (_vbd === undefined || _vbd.is_cd_drive) {
+      return acc
+    }
+
+    const vdi = getVdi(_vbd.VDI)
+
+    if (vdi === undefined) {
+      console.error('VDI not found')
+      return acc
+    }
+
+    acc.push({
+      name_label: vdi.name_label,
+      name_description: vdi.name_description,
+      size: bytesToGiB(vdi.size),
+      sr: vdi.$SR,
+      userdevice: _vbd.device,
```

Comment:
I know I told you to use  `vbd.device` but after rereading `xapi-object-to-xo` it's `position` and not `device` Sorry.
By the way, you need to add this property to your `xo-api-definition` otherwise it won't be retrieved.